### ISSUE TITLE
Campfire and candles inside a claim can be lit with fire arrows, without trust i the claim

### ISSUE THREAD

**Issue by snurre0** [GriefPrevention/GriefPrevention#1560]
### What happened:
Campfires and candles in a claim can be lit with no trust, by shooting them with a flame arrow.

### What was expected:
Not being able to light campfires and candles in claims without trust.

### Steps to reproduce:
1. Player one creates a claim with noone trusted
2. Player one place unlit candles and unlit campfires inside the claim.
3. Player two shoots the campfires and candles inside the claim, with fire arrows. Both will be lit.

### Server and GriefPrevention version:
Paste of `/version`: This server is running Paper version git-Paper-239 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: ca708a0)
Paste of `/version GriefPrevention`: GriefPrevention version 85b76cc

### Stack trace/error or server log
I have not included server log, as I don't want to expose the entire chat log here. If it's still needed, I'll see what I can do.

### GriefPrevention config.yml

```yml
GriefPrevention:
  SeaLevelOverrides:
    world: -1
    world_nether: -1
    world_the_end: -1
  Claims:
    Mode:
      world: Survival
      world_the_end: Survival
      world_nether: Survival
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: true
    LockTrapDoors: true
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 225
    Claim Blocks Accrued Per Hour:
      Default: 150
    Max Accrued Claim Blocks:
      Default: 300000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 20
    MinimumWidth: 1
    MinimumArea: 10
    MaximumDepth: 0
    InvestigationTool: STICK
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 9999999
      UnusedClaimDays: 9999999
      AllClaims:
        DaysInactive: 9999999
        ExceptWhenOwnerHasTotalClaimBlocks: 10000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: true
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local;/minecraft:me;/tpa;/suicide
    WhisperSlashCommands: ''
    WarningMessage: Please reduce your noise level.  Spammers will be muted.
    BanOffenders: false
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 60
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      world: true
      world_nether: true
      world_the_end: true
    ProtectFreshSpawns: true
    PunishLogout: false
    CombatTimeoutSeconds: 15
    AllowCombatItemDrop: true
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: false
      AdministrativeClaims: false
      AdministrativeSubdivisions: false
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: false
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: false
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: true
  Economy:
    ClaimBlocksMaxBonus: 0
    ClaimBlocksPurchaseCost: 0.0
    ClaimBlocksSellValue: 0.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: true
    NonPvPWorlds: true
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: false
  BlockSurfaceOtherExplosions: false
  LimitSkyTrees: false
  LimitTreeGrowth: false
  PistonMovement: EVERYWHERE
  PistonExplosionSound: true
  FireSpreads: true
  FireDestroys: true
  AdminsGetWhispers: false
  AdminsGetSignNotifications: true
  SmartBan: true
  Mute New Players Using Banned Words: true
  MaxPlayersPerIpAddress: 5
  SilenceBans: true
  Siege:
    Worlds: []
    BreakableBlocks:
    - DIRT
    - GRASS
    - LONG_GRASS     <-- can't understand this entry, see BukkitDev documentation
    - COBBLESTONE
    - GRAVEL
    - SAND
    - GLASS
    - THIN_GLASS     <-- can't understand this entry, see BukkitDev documentation
    - WOOD     <-- can't understand this entry, see BukkitDev documentation
    - WOOL     <-- can't understand this entry, see BukkitDev documentation
    - SNOW
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: true
  SilverfishBreakBlocks: true
  CreaturesTrampleCrops: true
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: true
  Database:
    URL: ''
    UserName: ''
    Password: ''
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: false
      Administrative Activity: true
      Debug: false
      Muted Chat Messages: true
```

### Plugin list (if applicable):
AntiXRay, AutoMessage, BuycraftX, Chunky, CommandHook, EntityWatch, Essentials, EssentialsChat, EssentialsDiscord, EssentialsSpawn, F3NPerm, GriefPrevention, HolographicDisplays, InteractiveBooks, InventoryRollback, LogBlock, LuckPerms, Matrix, OpenInv, PlugMan, ProtocolLib, Reports, Vault, VoteRoulette, Votifier, WorldEdit, WorldGuard

### Debugging steps attempted

- [ ] (If applicable) I attempted running the server without GriefPrevention installed
  - [ ] (If applicable) The problem does _not_ occur if GriefPrevention is removed from the server.
- [x] I attempted running only GriefPrevention on the server
  - [x] Issue still occurs even if GriefPrevention is the only plugin running
- [x] I attempted testing for the issue on a new server
  - [x] Issue still occurs even after testing on a new server
- [x] I attempted testing for the issue with the vanilla client (no mods)
  - [x] Issue still occurs even after testing with the vanilla Minecraft client

**Comment by Jikoo** [comment]
What a royal pain in the tush. Dove into this because I was the last one that touched it in #1423.

Turns out that these blocks specifically are inconsistent with every other block because they do not also fire an EntityChangeBlockEvent. Furthermore, we can't unify the logic in the BlockIgniteEvent instead because cancelling the event has inconsistent results for various blocks (looking at you, TNT). Really this should probably be considered a Spigot bug, EntityChangeBlockEvent should also be fired, but it's definitely possible to address. May need to be cautious to avoid doubling denial messages in certain cases.

**Comment by snurre0** [comment]
Not sure if I should make a separate issue about this, but candles can also be extinguished by rightclicking if you have containertrust, which I would expect to require full trust. Espesially since you can't light them with containertrust.

**Comment by Jikoo** [comment]
Hmm, probably a separate issue. I see how it happened - candle cakes are specifically a lower tier of trust because they are edible, so they follow the cake rules. Candles were then added to the same level of trust because it makes sense for candles to be in the same tier of trust as candle cakes... except overall it doesn't, because candles aren't edible. <sup><sup>Yet. Can't wait for the upcoming crayon eating mod expansion.</sup></sup>

**Comment by bigpresh** [comment]
> Not sure if I should make a separate issue about this, but candles can also be extinguished by rightclicking if you have containertrust, which I would expect to require full trust. Espesially since you can't light them with containertrust.

Yeah, I'd say separate issue for that - but It *looks* like this should be fixed already by 134729167b15d0dce8b4f5d7656ef6c19b9c7e3b - are you able to try a recent CI build (download from https://ci.appveyor.com/project/RoboMWM39862/griefprevention/build/artifacts ) and see if that solves it for you?

EDIT: this being the particular CI build from that PR: https://ci.appveyor.com/project/RoboMWM39862/griefprevention/builds/40448069/artifacts

(It looks like AppVeyor only runs builds from PRs, not from pushes/merges to master, which is a bit of a shame, so there's no "current state of master" build...)

**Comment by Jikoo** [comment]
> It looks like this should be fixed already by 1347291

As stated in my post, it actually incorrectly places them in the container trust category.
